include/nuttx/serial/uart_16550.h

```diff
@@ -173,35 +173,20 @@
 
 /* Register offsets *********************************************************/
 
-#define UART_RBR_INCR          0  /* (DLAB =0) Receiver Buffer Register */
-#define UART_THR_INCR          0  /* (DLAB =0) Transmit Holding Register */
-#define UART_DLL_INCR          0  /* (DLAB =1) Divisor Latch LSB */
-#define UART_DLM_INCR          1  /* (DLAB =1) Divisor Latch MSB */
-#define UART_IER_INCR          1  /* (DLAB =0) Interrupt Enable Register */
-#define UART_IIR_INCR          2  /* Interrupt ID Register */
-#define UART_FCR_INCR          2  /* FIFO Control Register */
-#define UART_LCR_INCR          3  /* Line Control Register */
-#define UART_MCR_INCR          4  /* Modem Control Register */
-#define UART_LSR_INCR          5  /* Line Status Register */
-#define UART_MSR_INCR          6  /* Modem Status Register */
-#define UART_SCR_INCR          7  /* Scratch Pad Register */
-#define UART_USR_INCR          31 /* UART Status Register */
-#define UART_DLF_INCR          48 /* Divisor Latch Fraction Register */
-
-#define UART_RBR_OFFSET        (CONFIG_16550_REGINCR*UART_RBR_INCR)
-#define UART_THR_OFFSET        (CONFIG_16550_REGINCR*UART_THR_INCR)
```

Comment:
why not set CONFIG_16550_REGINCR to default value?(1)

Reply:
on-chip 16550 device register increment (controlled by `CONFIG_16550_REGINCR`) can be different for PCI serial drivers (which is controlled by PCI serial driver implementation). Decoupling register offsets from `CONFIG_16550_REGINCR` allows the use of on-chip 16550 and PCI-based devices simultaneously.

Reply:
Perhaps we can remove the `priv->regincr` member variable to save data size, so that the register offset could be determined at compile time.

Reply:
UART16550 PCI driver can't know which increasement will be used before the probe finish.

Reply:
OK, I just noticed that the PCI driver is supported